src/unit-number/unit-number-input.tsx

```diff
@@ -0,0 +1,390 @@
+        _setFloorValue(data);
+    };
+
+    const setUnitValue = (data: string) => {
+        unitValueStateRef.current = data;
+        _setUnitValue(data);
+    };
+
+    const setCurrentFocus = (data: FieldType) => {
+        currentFocusStateRef.current = data;
+        _setCurrentFocus(data);
+    };
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+    useEffect(() => {
+        document.addEventListener("mousedown", handleMouseDown);
+
+        if (nodeRef.current) {
+            nodeRef.current.addEventListener("keydown", handleNodeKeyDown);
+        }
+
+        if (floorInputRef.current) {
+            floorInputRef.current.addEventListener("keydown", handleKeyDown);
+        }
+        if (unitInputRef.current) {
+            unitInputRef.current.addEventListener("keydown", handleKeyDown);
+        }
+
+        return () => {
+            document.removeEventListener("mousedown", handleMouseDown);
+
+            if (nodeRef.current) {
+                nodeRef.current.removeEventListener(
+                    "keydown",
+                    handleNodeKeyDown
+                );
+            }
+
+            if (floorInputRef.current) {
+                floorInputRef.current.removeEventListener(
+                    "keydown",
+                    handleKeyDown
+                );
+            }
+            if (unitInputRef.current) {
+                unitInputRef.current.removeEventListener(
+                    "keydown",
+                    handleKeyDown
+                );
+            }
+        };
+    }, [currentFocus]);
+
+    useEffect(() => {
+        // Auto focus feature
+        if (
+            currentFocus === "floor" &&
+            floorValue.length === 3 &&
+            unitInputRef.current
+        ) {
+            unitInputRef.current.focus();
+        }
+    }, [floorValue]);
+
+    useEffect(() => {
+        formatDisplayValues(value);
+    }, [value]);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const handleKeyDown = (event: KeyboardEvent) => {
+        /**
+         * NOTE: This is the most deterministic way in handling
+         * incorrect characters from being entered. The pattern
+         * added in the input still allows some special characters
+         * to slip through.
+         */
+
+        const permittableEventCodes = [
+            "Tab",
+            "Backspace",
+            "Delete",
+            "ArrowLeft",
+            "ArrowRight",
+        ];
+
+        // form validation for inputs is not supported on ios,
+        // hence manual checks needed to prevent special characters input
+        const isPermittedCharacter =
+            /[0-9A-Za-z]/.test(event.key) ||
```

Comment:
was able to enter special characters on Android browser, event.key was `Unidentified`